### ISSUE TITLE
rebase: do not try to `fixup` a skipped commit

### ISSUE THREAD

**Issue by dscho** [gitgitgadget/git#1321]
When running an interactive rebase, it is possible that the target commit of a `fixup!` commit could not be applied due to merge conflicts (or because it has been upstreamed). In this instance, going forward with a `fixup` will amend the wrong commit.

Let's prevent that.

The specifics how to prevent that are not set in stone. The easiest solution would probably be to refuse to even start with the `fixup` command, to complain loudly, and to re-issue the command for the user to edit.

The tricky part will be to detect that situation because the target commit might have failed to apply cleanly, requiring manual fixing up followed by `git add -u && git rebase --continue`, but those manual fixups might have resulted in no more changes to be committed, either.

**Comment by dscho** [comment]
See https://github.com/git-for-windows/git/pull/3981#issuecomment-1212856115 for a real-world scenario where this bug caused real problems.

**Comment by derrickstolee** [comment]
By "refusing to start the `fixup!`" do you instead mean to keep the commit as a `pick` and not reorder it?

**Comment by dscho** [comment]
No, at the time we could reorder it, we won't have the information whether its target `pick` caused problems and was skipped by the user.

**Comment by samkergachu** [comment]
Hey dscho. Is anyone working on this, I would like to take it up as my outreachy microproject

**Comment by vinayakdsci** [comment]
Hey @dscho! Could you help me take up this issue as my GSoC microproject?
I would be very grateful.

**Comment by dscho** [comment]
> Hey @dscho! Could you help me take up this issue as my GSoC microproject? I would be very grateful.

Unfortunately, I am pressed for time. You could reach out to the Git mailing list for guidance, though. You can maximize your chances by being specific about your needs.

**Comment by vinayakdsci** [comment]
Absolutely, I can understand. I will ask on the mailing list. Thanks!

**Comment by Cheskaqiqi** [comment]
Hey dscho. Is anyone still working on this.I would like to have a try .

**Comment by Cheskaqiqi** [comment]
Hey dscho. Is still anyone working on this . I would like to have a try.

**Comment by vinayakdsci** [comment]
Hey Cheskaqiqi!
Yes, I am on it.

**Comment by AyuSharma176** [comment]
Can i get information about this issue i am new contributor want some information about this issue
